pkg/crd/schema.go

```diff
@@ -291,6 +291,12 @@ func mapToSchema(ctx *schemaContext, mapType *ast.MapType) *v1beta1.JSONSchemaPr
 		valSchema = localNamedToSchema(ctx.ForInfo(&markers.TypeInfo{}), val)
 	case *ast.SelectorExpr:
 		valSchema = namedToSchema(ctx.ForInfo(&markers.TypeInfo{}), val)
+	case *ast.ArrayType:
+		valSchema = arrayToSchema(ctx.ForInfo(&markers.TypeInfo{}), val)
+		if valSchema.Type != "string" {
```

Comment:
this needs a comment on why string works here, since it's not immediately obvious

Reply:
also we'll want to add support for string slices, as per an addendum to the issue.

Reply:
> this needs a comment on why string works here, since it's not immediately obvious
Would you mind of sharing the reason why string works here? Thanks!